File: algorithm/All-In-One-Java/문제/week5/src/Problem1.java
/**
 * Problem 1: 두 정수 사이의 합
 * 난이도: 🟢 Lv.1
 * 주제: 수학, 루프
 *
 * 문제: 두 정수 a, b가 주어질 때, a와 b 사이에 속한 모든 정수의 합을 반환하세요.
 *      (a == b 인 경우 a 반환, a > b 인 경우에도 동작해야 함)
 *
 * 입력: a = 3, b = 5
 * 출력: 12  (3 + 4 + 5)
 *
 * 핵심: 등차수열의 합 공식 또는 루프로 계산
 */

public class Problem1 {

    static class Solution {
        public long solution(int a, int b) {
            // TODO: 풀이 작성
            return 0L;
        }
    }

    public static void main(String[] args) {
        Solution sol = new Solution();

        System.out.println("=== Problem 1: 두 정수 사이의 합 ===\n");

        int[][] inputs = {
                { 3, 5 },
                { 3, 3 },
                { 5, 3 },
                { -3, 3 },
                { -5, -3 }
        };
        long[] expecteds = { 12L, 3L, 12L, 0L, -12L };

        for (int i = 0; i < inputs.length; i++) {
            int a = inputs[i][0];
            int b = inputs[i][1];
            long expected = expecteds[i];
            long result = sol.solution(a, b);
            String status = expected == result ? "✓" : "✗";

            System.out.println("Test " + (i + 1) + ":");
            System.out.println("  a = " + a + ", b = " + b);
            System.out.println("  결과: " + result + " (기대값: " + expected + ") " + status + "\n");
        }
    }
}

File: algorithm/All-In-One-Java/문제/week6/src-js/Problem1.js
/**
 * Problem 1: 평균 구하기
 * 난이도: 🟢 Lv.1
 * 주제: 배열, 기본 연산
 *
 * 문제: 정수를 담고 있는 배열 arr의 평균값을 return하는 함수를 작성하세요.
 *
 * 입력: arr = [1, 2, 3, 4]
 * 출력: 2.5
 *
 * 핵심: 합계 구하기, 길이로 나누기
 */

function solution(arr) {
  // TODO: 풀이 작성
}

// === 테스트 ===
const tests = [
  { input: [[1, 2, 3, 4]], expected: 2.5 },
  { input: [[5, 5]], expected: 5 },
  { input: [[10]], expected: 10 },
  { input: [[-1, 0, 1]], expected: 0 },
];

console.log("=== Problem 1: 평균 구하기 ===\n");

tests.forEach((t, i) => {
  const [arr] = t.input;
  const result = solution(arr);
  const ok = result === t.expected;
  console.log(
    `Test ${i + 1}: [${arr}] → ${result} (기대값: ${t.expected}) ${ok ? "✓" : "✗"}`
  );
});
